resource-aggregate/pb/commands.proto

```diff
@@ -385,3 +385,96 @@ message ConfirmResourceCreateResponse {
+//     │ │                   │                                      │                              │                       │ │  
+//     │ │                  ┌┴┐                     DeviceMetadataUpdatePending                    │                       │ │  
+//     │ │                  │ │ <─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─                       │ │  
+//     │ │                  │ │                                     │                              │                       │ │  
+//     │ │                  │ │ ConfirmDeviceMetadataUpdateRequest  ┌┴┐                            │                       │ │  
+//     │ │                  │ │ ───────────────────────────────────>│ │                            │                       │ │  
+//     │ │                  └┬┘                                     │ │                            │                       │ │  
+//     │ │                   │ConfirmDeviceMetadataUpdateResponse   │ │                            │                       │ │  
+//     │ │                   │<──────────────────────────────────── │ │                            │                       │ │  
+//     │ │                   │                                      └┬┘                            │                       │ │  
+//     │ │                   │                                      │    DeviceMetadataUpdated     │                       │ │  
+//     │ │                   │                                      │  ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─>│                       │ │  
+//     └┬┘                   │                                      │                              │                       └┬┘  
+//      │                    │                                      │                              │ DeviceMetadataUpdated │    
+//      │                    │                                      │                              │  ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─>    
+// ┌──────────┐          ┌───┴───┐                         ┌────────┴─────────┐               ┌─────────┐               ┌──────┐
+// │OCF Server│          │Gateway│                         │Resource Aggregate│               │Event Bus│               │Client│
+// └──────────┘          └───────┘                         └──────────────────┘               └─────────┘               └──────┘
+message ShadowSynchronizationStatus {
+    bool disabled = 1;
+}
+
+message UpdateDeviceMetadataRequest {
+    string device_id = 1;
+    string correlation_id = 2;
+    oneof update {
+        ConnectionStatus status = 3;
+        ShadowSynchronizationStatus shadow_synchronization_status = 4;
+    };
+    CommandMetadata command_metadata = 100;
+}
+
+message UpdateDeviceMetadataResponse {
+    AuditContext audit_context = 100;
+}
+
+message ConfirmDeviceMetadataUpdateRequest {
+    string device_id = 1;
+    string correlation_id = 2;
+    oneof confirm {
+        ShadowSynchronizationStatus shadow_synchronization_status = 3;
```

Comment:
Why need to wrap it? If it's one of, it cannot be directly bool?

Reply:
No. Maybe in future we want to add time how long it will be disabled ...